Random network panic in 1.7

I'm running f83073d3eb23a3f3c22bef9401090df5da53cc2f with `docker -d -b none`.  On container start it does this.

```
INFO[0001] POST /containers/create?hostconfig=%7B%22Binds%22%3A%5B%22%2Fdev%3A%2Fhost%2Fdev%22%2C%22%2Flib%2Fmodules%3A%2Flib%2Fmodules%22%2C%22%2Flib%2Ffirmware%3A%2Flib%2Ffirmware%22%5D%2C%22Privileged%22%3Atrue%2C%22NetworkMode%22%3A%22host%22%2C%22RestartPolicy%22%3A%7B%7D%2C%22LogConfig%22%3A%7B%22Type%22%3A%22json-file%22%7D%7D&name=udev 
INFO[0001] POST /containers/3093e640d9666474febb6616b5a1d3dcd87993b04612ab685d2833079b28f274/start 
WARN[0001] IPv4 forwarding is disabled. Networking will not work 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x4c8380]

goroutine 133 [running]:
github.com/docker/docker/daemon.(*Container).ReleaseNetwork(0xc2081001e0)
	/go/src/github.com/docker/docker/daemon/container_linux.go:913 +0xc0
github.com/docker/docker/daemon.(*Container).cleanup(0xc2081001e0)
	/go/src/github.com/docker/docker/daemon/container.go:338 +0x33
github.com/docker/docker/daemon.(*containerMonitor).Close(0xc2081879d0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/monitor.go:86 +0x4a
github.com/docker/docker/daemon.func·031()
	/go/src/github.com/docker/docker/daemon/monitor.go:117 +0xbe
github.com/docker/docker/daemon.(*containerMonitor).Start(0xc2081879d0, 0x7ffad7192f40, 0xc20821dfa0)
	/go/src/github.com/docker/docker/daemon/monitor.go:145 +0x4f6
github.com/docker/docker/daemon.*containerMonitor.Start·fm(0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/container.go:703 +0x39
github.com/docker/docker/pkg/promise.func·001()
	/go/src/github.com/docker/docker/pkg/promise/promise.go:8 +0x2f
created by github.com/docker/docker/pkg/promise.Go
	/go/src/github.com/docker/docker/pkg/promise/promise.go:9 +0xfb
```

--- a/daemon/container_linux.go
+++ b/daemon/container_linux.go
@@ -182,17 +182,20 @@ func getDevicesFromPath(deviceMapping runconfig.DeviceMapping) (devs []*configs.
 }
 
 func populateCommand(c *Container, env []string) error {
-	en := &execdriver.Network{
-		NamespacePath: c.NetworkSettings.SandboxKey,
-	}
-
-	parts := strings.SplitN(string(c.hostConfig.NetworkMode), ":", 2)
-	if parts[0] == "container" {
-		nc, err := c.getNetworkedContainer()
-		if err != nil {
-			return err
+	var en *execdriver.Network
+	if !c.daemon.config.DisableNetwork {
+		en = &execdriver.Network{
+			NamespacePath: c.NetworkSettings.SandboxKey,
 		}
-		en.ContainerID = nc.ID
+
+		parts := strings.SplitN(string(c.hostConfig.NetworkMode), ":", 2)
+		if parts[0] == "container" {
+			nc, err := c.getNetworkedContainer()
+			if err != nil {
+				return err
+			}
+			en.ContainerID = nc.ID
+		}
 	}
 
 	ipc := &execdriver.Ipc{}
@@ -906,7 +909,7 @@ func (container *Container) getNetworkedContainer() (*Container, error) {
 }
 
 func (container *Container) ReleaseNetwork() {
-	if container.hostConfig.NetworkMode.IsContainer() {
+	if container.hostConfig.NetworkMode.IsContainer() || container.daemon.config.DisableNetwork {
 		return
 	}
 

--- a/daemon/execdriver/native/create.go
+++ b/daemon/execdriver/native/create.go
@@ -89,6 +89,9 @@ func generateIfaceName() (string, error) {
 }
 
 func (d *driver) createNetwork(container *configs.Config, c *execdriver.Command) error {
+	if c.Network == nil {
+		return nil
+	}
 	if c.Network.ContainerID != "" {
 		d.Lock()
 		active := d.activeContainers[c.Network.ContainerID]

--- a/integration-cli/docker_cli_daemon_test.go
+++ b/integration-cli/docker_cli_daemon_test.go
@@ -1156,3 +1156,12 @@ func (s *DockerDaemonSuite) TestCleanupMountsAfterCrash(c *check.C) {
 	c.Assert(err, check.IsNil, check.Commentf("Output: %s", mountOut))
 	c.Assert(strings.Contains(string(mountOut), id), check.Equals, false, check.Commentf("Something mounted from older daemon start: %s", mountOut))
 }
+
+func (s *DockerDaemonSuite) TestRunContainerWithBridgeNone(c *check.C) {
+	c.Assert(s.d.StartWithBusybox("-b", "none"), check.IsNil)
+
+	out, err := s.d.Cmd("run", "--rm", "busybox", "ip", "l")
+	c.Assert(err, check.IsNil, check.Commentf("Output: %s", out))
+	c.Assert(strings.Contains(out, "eth0"), check.Equals, false,
+		check.Commentf("There shouldn't be eth0 in container when network is disabled: %s", out))
+}

